source/common/formatter/stream_info_formatter.cc

```diff
@@ -1117,37 +1164,37 @@ const StreamInfoFormatterProviderLookupTable& getKnownStreamInfoFormatterProvide
           {"UPSTREAM_REMOTE_ADDRESS",
            {CommandSyntaxChecker::COMMAND_ONLY,
             [](const std::string&, absl::optional<size_t>) {
+              const bool use_connection_address = Runtime::runtimeFeatureEnabled(
+                  "envoy.reloadable_features.upstream_remote_address_use_connection");
+
               return StreamInfoAddressFormatterProvider::withPort(
-                  [](const StreamInfo::StreamInfo& stream_info)
-                      -> std::shared_ptr<const Envoy::Network::Address::Instance> {
-                    if (stream_info.upstreamInfo() && stream_info.upstreamInfo()->upstreamHost()) {
-                      return stream_info.upstreamInfo()->upstreamHost()->address();
-                    }
-                    return nullptr;
+                  [use_connection_address](const StreamInfo::StreamInfo& stream_info)
+                      -> Envoy::Network::Address::InstanceConstSharedPtr {
+                    return getUpstreamRemoteAddress(stream_info, use_connection_address);
```

Comment:
As written, it seems to me that changing the runtime feature value has no effect until the access log format is re-parsed. Was that intentional? 

I think the behavior might be surprising given that most flags have immediate effect. I think we should either change the behavior so the guard is checked when the formatter is invoked (e.g. immediate effect) or else document that toggling the flag requires a config change to trigger the old behavior.

Reply:
Yeah, it is intentional to avoid doing flag searching every time.
Fine, let's make it take effect immediately, it will be removed anyway after six month.

Reply:
done.